catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/inputtransformer/InputTransformerProducer.java

```diff
@@ -130,64 +137,39 @@ private Metacard generateMetacard(
-        throw new MetacardCreationException("Could not copy bytes of content message.", e);
-      }
-
-      // Multiple InputTransformers may be found that match the mime type.
-      // Need to try each InputTransformer until we find one that can successfully transform
-      // the input stream's data into a metacard. Once an InputTransformer is found that
-      // can create the metacard, then do not need to try any remaining InputTransformers.
-      for (InputTransformer transformer : listOfCandidates) {
-
-        try (InputStream inputStreamMessageCopy =
-            fileBackedOutputStream.asByteSource().openStream()) {
-          if (StringUtils.isEmpty(metacardId)) {
-            generatedMetacard = transformer.transform(inputStreamMessageCopy);
-          } else {
-            generatedMetacard = transformer.transform(inputStreamMessageCopy, metacardId);
-          }
-        } catch (IOException | CatalogTransformerException e) {
-          LOGGER.debug("Transformer [{}] could not create metacard.", transformer, e);
-        }
-        if (generatedMetacard != null) {
-          break;
+    // Multiple InputTransformers may be found that match the mime type.
+    // Need to try each InputTransformer until we find one that can successfully transform
+    // the input stream's data into a metacard. Once an InputTransformer is found that
+    // can create the metacard, then do not need to try any remaining InputTransformers.
+    for (InputTransformer transformer : listOfCandidates) {
+
+      try (InputStream inputStreamMessageCopy = tfbos.asByteSource().openStream()) {
+        if (StringUtils.isEmpty(metacardId)) {
+          generatedMetacard = transformer.transform(inputStreamMessageCopy);
+        } else {
+          generatedMetacard = transformer.transform(inputStreamMessageCopy, metacardId);
         }
+      } catch (CatalogTransformerException e) {
+        LOGGER.debug("Transformer [{}] could not create metacard.", transformer, e);
+      } catch (IOException e) {
+        LOGGER.debug("Could not open input stream", e);
       }
-
-      if (generatedMetacard == null) {
-        throw new MetacardCreationException(
-            "Could not create metacard with mimeType "
-                + mimeType
-                + ". No valid transformers found.");
+      if (generatedMetacard != null) {
+        break;
       }
-
-      LOGGER.trace("EXITING: generateMetacard");
-    } catch (IOException e) {
-      throw new MetacardCreationException("Could not create metacard.", e);
     }
 
-    return generatedMetacard;
+    LOGGER.trace("EXITING: generateMetacard");
+    return Optional.ofNullable(generatedMetacard);
   }
 
-  private String getMimeTypeFromHeader(Message in, InputStream is) {
-    String fileExtension = getHeaderAsStringAndRemove(in, FILE_EXTENSION_HEADER);
-    if (fileExtension == null) {
-      return null;
-    }
-
+  private Optional<String> getMimeTypeFor(InputStream is, String fileExtension) {
```

Comment:
✏️ Returning an optional here didn't simplify the code since we are now forced above to check if it is present and then do a get which ends up checking if it is present twice anyway. Three times if you can the other get done inside the trace.